default/python/AI/turn_state/_supply_state.py

```diff
@@ -72,3 +73,49 @@ def _get_enemy_supply_distance_map() -> Mapping[int, int]:
 
 def get_distance_to_enemy_supply(sys_id: int) -> int:
     return _get_enemy_supply_distance_map().get(sys_id, 999)
+
+
+@cache_for_current_turn
+def _system_to_supply_group() -> Mapping[SystemId, int]:
+    """
+    Create a mapping of SystemIds to supply_group_id.
+    Used buy supply_connected, get_supply_group_id and get_supply_group.
+    """
+    result = {}
+    num = 0
+    for group in fo.getEmpire().resourceSupplyGroups:
+        num += 1
+        result.update((sys_id, num) for sys_id in group)
+    return result
+
+
+def supply_connected(system_id1: SystemId, system_id2: SystemId) -> bool:
+    """Return whether the two systems are connected by this empire's supply chains."""
+    mapping = _system_to_supply_group()
+    # Different defaults to get False when neither system is part of any group
+    return mapping.get(system_id1, -1) == mapping.get(system_id2, -2)
+
+
+def get_supply_group_id(system_id: SystemId) -> int:
```

Comment:
Maybe we could hide this from the API and move to the get_supply group, which will accept system_id?

Reply:
At the moment I do use get_supply_group_id in _rate_supply_extension and it would not be trivial to replace it